File: src/main/java/ModelsHandler.java
// ModelsHandler.java
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import org.json.JSONObject;

import java.io.IOException;
import java.io.OutputStream;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.List;

public class ModelsHandler implements HttpHandler {
    private static final List<JSONObject> models = new ArrayList<>();
    static{
        models.add(new JSONObject().put("id", "gpt-4o-2024-08-06").put("object", "model"));

    }
    @Override
    public void handle(HttpExchange exchange) throws IOException {

        // 仅允许GET请求
        if (!exchange.getRequestMethod().equalsIgnoreCase("GET")) {
            exchange.sendResponseHeaders(405, -1); // Method Not Allowed
            return;
        }

        try {
            // 构建响应JSON
            JSONObject responseJson = new JSONObject();
            responseJson.put("object", "list");
            responseJson.put("data", models);

            byte[] responseBytes = responseJson.toString().getBytes(StandardCharsets.UTF_8);
            exchange.getResponseHeaders().add("Content-Type", "application/json");
            exchange.sendResponseHeaders(200, responseBytes.length);

            OutputStream os = exchange.getResponseBody();
            os.write(responseBytes);
            os.close();
            System.gc();
        } catch (Exception e) {
            e.printStackTrace();
            exchange.sendResponseHeaders(500, -1);
        }
    }
}

File: src/main/cpp/models_handler.cpp
#include "utils.h"
#include <nlohmann/json.hpp>
#include <httplib.h>

using json = nlohmann::json;

// C++ implementation of Java's ModelsHandler
// Exposes a GET /models endpoint that returns a list of available models.
void handleModels(const httplib::Request& /*req*/, httplib::Response& res) {
    try {
        json model;
        model["id"] = "gpt-4o-2024-08-06";
        model["object"] = "model";

        json resp;
        resp["object"] = "list";
        resp["data"] = json::array({ model });

        std::string body = resp.dump();
        res.set_header("Content-Type", "application/json; charset=utf-8");
        res.status = 200;
        res.body = body;
    } catch (const std::exception& e) {
        Utils::send_error(res, std::string("Models handler error: ") + e.what(), 500);
    }
}
